### PR TITLE
CODENVY-560 Adapt changes to new management of MachineSource in Che

### PR DESCRIPTION
**Pull request by benoitf**
where InstanceKey class is removed as it can use the MachineSource object

Also update to use new Constructor of MachineSource

requires https://github.com/eclipse/che/pull/1366

Change-Id: I94a9705eabe1ea4fa750e1fafb1864c5df67b6f5
Signed-off-by: Florent BENOIT <fbenoit@codenvy.com>

### CODE REVIEW

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/SnapshotImplCodec.java [hunk 1]
```diff
@@ -66,6 +72,13 @@ public void encode(BsonWriter writer, SnapshotImpl snapshot, EncoderContext enco
     @Override
     public SnapshotImpl decode(BsonReader reader, DecoderContext decoderContext) {
         final Document document = codec.decode(reader, decoderContext);
+
+        final Document machineSourceDocument = (Document)document.get(MACHINE_SOURCE);
+
+        MachineSourceImpl machineSource =
+                new MachineSourceImpl(machineSourceDocument.getString("type")).setLocation(machineSourceDocument.getString("location"))
```
Comment (garagatyi) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/SnapshotImplCodec.java]: Either use constants you've introduced or remove them

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java [hunk 2]
```diff
@@ -239,7 +239,7 @@ private static MachineConfigImpl asMachineConfig(Document document) {
                                                                   .setType(document.getString("type"));
         final Document sourceDocument = document.get("source", Document.class);
         if (sourceDocument != null) {
-            builder.setSource(new MachineSourceImpl(sourceDocument.getString("type"), sourceDocument.getString("location")));
+            builder.setSource(new MachineSourceImpl(sourceDocument.getString("type")).setLocation(sourceDocument.getString("location")));
```
Comment (garagatyi) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java]: Have you forgot content?

Comment (benoitf) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java]: fixed

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/SnapshotImplCodec.java (hunk 1 above)
Comment (benoitf) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/SnapshotImplCodec.java]: fixed

**Comment by benoitf** [comment]
@garagatyi @skabashnyuk of for you now on this repository ?

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java [hunk 3]
```diff
@@ -239,7 +239,9 @@ private static MachineConfigImpl asMachineConfig(Document document) {
                                                                   .setType(document.getString("type"));
         final Document sourceDocument = document.get("source", Document.class);
         if (sourceDocument != null) {
-            builder.setSource(new MachineSourceImpl(sourceDocument.getString("type"), sourceDocument.getString("location")));
+            builder.setSource(new MachineSourceImpl(sourceDocument.getString("type"))
+                                      .setContent(sourceDocument.getString("content"))
+                                      .setLocation(sourceDocument.getString("location")));
```
Comment (garagatyi) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java]: Following your usage of constants for fields consider moving **MACHINE_SOURCE...** constants to this class and usage them in this class and SnapshotImplCodec

**Comment by garagatyi** [comment]
LGTM

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java (hunk 3 above)
Comment (benoitf) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceImplCodec.java]: yes fixed

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**